internal/gps/lock.go

```diff
@@ -195,6 +196,11 @@ func (lp LockedProject) Packages() []string {
 	return lp.pkgs
 }
 
+func (lp LockedProject) String() string {
+	return fmt.Sprintf("%s@%v for packages: %v",
```

Comment:
hmm. so, we've a pair of problems.

first is that this is a generic `Stringer` implementation, but word choice is oriented towards the one use case we currently have: using the preposition "for" makes the most sense as a sort of explanation "we have this project for packages X, Y, Z within it." but that seems less than ideal, in part because that list of packages includes ones that are only required internally (e.g., some other project imports `root/A`, which imports its sibling `root/B`; both show up in the `packages` list.) to my mind, "with" is a bit better than "for", because it's more of a plain statement/less expository.

however, there's also the problem that listing out the packages _at all_ probably gives the wrong impression to the user - it suggests that we're writing out only those packages, when we're actually writing out the entire tree.

so, let's use an unexported, purpose-specific func for generating this string, and omit package information from it entirely. basically, just the `"%s@%v"` part.

however, since we've already been through all this with the `LockedProject.String()` method, i'm ok with leaving it in (even though we're not using it), as long as we s/for/with/.